From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 09 Jan 2022 22:24:18 +0100
Subject: [PATCH] Use system-provided GTest

This also fixes the problem that `cmake --install` installs even GTest files.

See-Also: https://github.com/jbeder/yaml-cpp/commit/5e9cb0128d1b600ff74505429269fd4dff367f4a#r63176586
See-Also: https://github.com/jbeder/yaml-cpp/pull/1035

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -4,12 +4,8 @@
 set(BUILD_MOCK ON CACHE BOOL "" FORCE)
 set(CMAKE_POLICY_DEFAULT_CMP0048 NEW)
 
-add_subdirectory(
-  "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.10.0"
-  "${CMAKE_CURRENT_BINARY_DIR}/prefix")
+find_package(GTest REQUIRED)
 
-include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.10.0/googletest/include")
-
 set(test-new-api-pattern "new-api/*.cpp")
 set(test-source-pattern "*.cpp" "integration/*.cpp" "node/*.cpp")
 if (CMAKE_VERSION VERSION_GREATER 3.11)
@@ -38,7 +34,8 @@
   PRIVATE
     Threads::Threads
     yaml-cpp
-    gmock)
+    GTest::gtest
+    GTest::gmock)
 
 set_property(TARGET yaml-cpp-tests PROPERTY CXX_STANDARD_REQUIRED ON)
 if (NOT DEFINED CMAKE_CXX_STANDARD)
